src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/AddAccessDecisionStrategyPass.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Symfony\Bundle\SecurityBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class AddAccessDecisionStrategyPass implements CompilerPassInterface
+{
+    /**
+     * You can modify the container here before it is dumped to PHP code.
+     *
+     * @param ContainerBuilder $container
+     *
+     * @api
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('security.access.decision_manager')) {
+            return;
+        }
+
+        $strategies = array();
+        foreach ($container->findTaggedServiceIds('security.access_strategy') as $id => $attributes) {
```

Comment:
wrong naming. ``$attributes`` is not an array of attributes. It is an array of tags.

And you should support having several ``security.access_strategy`` tags on the same service by looping over tags instead of reading only the first one